src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -12,13 +13,24 @@ namespace Microsoft.DotNet.NativeWrapper
     public static partial class Interop
     {
         public static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+#if NETCOREAPP
+        private static readonly string HostFxrPath;
+#endif
 
         static Interop()
         {
             if (RunningOnWindows)
             {
-                PreloadWindowsLibrary("hostfxr.dll");
+                PreloadWindowsLibrary(Constants.HostFxr);
             }
+#if NETCOREAPP
+            else
+            {
+                HostFxrPath = (string)AppContext.GetData(Constants.RuntimeProperty.HostFxrPath);
+                if (!string.IsNullOrEmpty(HostFxrPath))
+                    NativeLibrary.SetDllImportResolver(typeof(Interop).Assembly, HostFxrDllImportResolver);
```

Comment:
We might want to fail if he hostfxr path is not available. Mainly because we know it is somewhat random if it will work without the resolver. In short we think this should never happen, but it's better to have diagnostics for this.